Framework/DataObjects/inc/MantidDataObjects/TableColumn.h

```diff
@@ -162,7 +162,17 @@ template <class Type> class TableColumn : public API::Column {
         typename boost::mpl::if_c<boost::is_convertible<double, Type>::value,
                                   Type, InconvertibleToDoubleType>::type
             DoubleType;
-    return boost::numeric_cast<double, DoubleType>(m_data[i]);
+    if (typeid(m_data[i]) == typeid(std::string)) {
+      std::string trimmedString = boost::lexical_cast<std::string>(
+          m_data[i]); // so not to alter the original data
+
+      if (char(trimmedString.back()) == ':') {
```

Comment:
I will double check, but if I recall correctly there was an error about comparing a string to a char.